sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
Can you have a look now? I wrapped everything with `ifdef`s for different OS:
https://github.com/leoleovich/server/blob/10.2/sql/sql_connect.cc#L1005-L1026